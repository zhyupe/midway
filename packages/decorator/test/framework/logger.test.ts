import {
  Logger,
  getClassMetadata,
  INJECT_CUSTOM_TAG,
} from '../../src';

class Test {
  @Logger()
  logger: any;

  @Logger('bbb')
  bbb: any;
}

describe('/test/framework/logger.test.ts', () => {
  it('logger decorator should be ok', () => {
    let data = getClassMetadata(INJECT_CUSTOM_TAG, Test);
    expect(data).toMatchSnapshot();
  });
});
