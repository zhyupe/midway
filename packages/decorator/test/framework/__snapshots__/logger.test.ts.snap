// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/logger.test.ts logger decorator should be ok 1`] = `
Object {
  "bbb": Object {
    "key": "logger",
    "propertyName": "bbb",
    "targetKey": "bbb",
  },
  "logger": Object {
    "key": "logger",
    "propertyName": "logger",
    "targetKey": "logger",
  },
}
`;
